File: SmartBat/BTSubdivision.m
//
//  BTBeat.m
//  SmartBat
//
//  Created by poppy on 13-6-6.
//  Copyright (c) 2013年 kaka'. All rights reserved.
//

#import "BTSubdivision.h"

@implementation BTSubdivision


-(BTSubdivision *) initWithBeat: (NSArray *) _beatDescription
{
    self = [super init];
    
    _noteList = [[NSArray alloc]initWithArray:_beatDescription];

    self.playIndex = 0;
    
    return self;

}

-(int)count
{
    return [_noteList count];
}

-(void) playNote
{
    if(self.playIndex == [_noteList count]-1){
        
        self.playIndex = 0;
    }
    else
    {
        self.playIndex ++;
    }
}

-(void)reset
{
    self.playIndex = 0;
}

-(BTBeatType)getNote: (int)index
{
    
    BTBeatType type = [[_noteList objectAtIndex:index] intValue];
    
    return type;
}

@end

File: java/src/com/sensormusic/metronomecore/BTSubdivision.java
package com.sensormusic.metronomecore;

import java.util.ArrayList;

public class BTSubdivision {

	public int playIndex;

	private ArrayList<BTBeat> _noteList;

	public void init(ArrayList<BTBeat> beatDescription) {

		this._noteList = beatDescription;
		this.playIndex = 0;
	}

	public BTBeat getCurrentNote() {
		BTBeat beat = this._noteList.get(this.playIndex);
		return beat;
	}

	public void playNote() {

	    if(this.playIndex == count()-1){
	        
	        this.playIndex = 0;
	    }
	    else
	    {
	        this.playIndex ++;
	    }
	}
	
	public void reset(){
		this.playIndex = 0;
	}
	
	public int count(){
		return this._noteList.size();
	}
}
